extensions/sasl-3.1.md

```diff
@@ -42,16 +42,21 @@ is received, the IRCD will attempt to establish an association with a SASL
 agent.) If this fails, a 904 numeric will be sent and the session state remains
 unchanged; the client MAY try another mechanism. Otherwise, the server sends
 a set of regular AUTHENTICATE messages with the initial server response.
+If the chosen mechanism is client-first, the server sends an empty response
+(`AUTHENTICATE +`, as described below).
 
     initial-authenticate = "AUTHENTICATE" SP mechanism CRLF
 
 A set of regular AUTHENTICATE messages transmits a response from client to
 server or vice versa. The server MAY intersperse other IRC protocol messages
-between the AUTHENTICATE messages of a set. The "+" form is used for an empty
-response. The server MAY place a limit on the total length of a response.
+between the AUTHENTICATE messages of a set.
+The response is encoded in Base64
+([RFC 4648](https://tools.ietf.org/html/rfc4648)), then split to 400-byte
+chunks. If length of response divides by 400 (e.g. if the response is empty), an
+additional `AUTHENTICATE +` message is transmitted in the end.
```

Comment:
Every time I re-read this, it seems _more_ confusing than before. Yes, `0 % 400 == 0`, but it doesn't seem to translate well.

Regardless of the above, I think it should also be clarified that the length is of the _encoded_ response...

How about:

> ... then split to 400-byte chunks, and each chunk is sent as a separate `AUTHENTICATE...` command. Empty (zero-length) responses are sent as `AUTHENTICATE +`. If the last chunk was exactly 400 bytes long, it must also be followed by `AUTHENTICATE +` to signal end of response.